linera-execution/src/execution.rs

```diff
@@ -216,15 +216,27 @@ where
                 contract.execute_message(context, runtime_sender, message)
             }
         });
-        runtime_actor.run().await?;
+        let runtime_result = runtime_actor.run().await;
         let call_result = call_result_future.await?;
 
         // TODO(#989): Make user errors fail blocks again.
-        let mut result = if let Err(ExecutionError::UserError(message)) = &call_result {
-            tracing::error!("Ignoring error reported by user application: {message}");
-            RawExecutionResult::default()
-        } else {
-            call_result?
+        let mut result = match (runtime_result, call_result) {
+            (
+                Err(ExecutionError::UserError(message)),
+                Err(ExecutionError::MissingRuntimeResponse),
```

Comment:
Do we need to check the `ExecutionError::MissingRuntimeResponse` ?

Reply:
One could use `_` for the second element of the pair then. Optionally, it could also be `Err(error)` together with `assert!(matches(error, ExecutionError::MissingRuntimeResponse))`.